src/main/java/appeng/container/guisync/GuiSync.java

```diff
@@ -19,11 +19,18 @@
 package appeng.container.guisync;
 
 
+import java.lang.annotation.ElementType;
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
 
 
+/**
+ * Annotates that this field should be synchronized between the server and client.
+ * Requires the field to be public.
+ */
 @Retention( RetentionPolicy.RUNTIME )
+@Target( ElementType.FIELD )
 public @interface GuiSync
```

Comment:
always wondered, why do we use self-allocated IDs? wouldnt an enum be better for the GUI ID?